UI: Fix scene tree item spacing

cg2121 opened the pull request:
### Description
The setSpacing was always set to 0 in the resize event.

Before:
![Screenshot from 2022-08-27 23-26-36](https://user-images.githubusercontent.com/19962531/187057602-b55648c3-6e5e-4cd0-bbe0-26bba3a37473.png)

After:
![Screenshot from 2022-08-29 17-09-34](https://user-images.githubusercontent.com/19962531/187307732-aeec2d0b-d224-49fb-80a7-ef72c7a735c3.png)

### Motivation and Context
Consistent UI

### How Has This Been Tested?
Looked at scene tree to make sure spacing was correct.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

WizardCM commented:
While this does fix spacing, heights between Sources and Scenes are different (scenes are a total of 2px taller in Yami)

![image](https://user-images.githubusercontent.com/941350/187210885-6c81a4ea-25cf-41a4-89fc-9a9ced21820d.png)

Warchamp7 commented:
Looking at this, the items in the SceneTree list are also 4 pixels off the top and sides versus 5 pixels in the SourceTree. I feel like there is another underlying problem causing this that we should investigate, rather than a bandaid in the stylesheet

cg2121 committed:
UI: Fix scene list spacing

The reason why the scene list spacing would never work is because
spacing was always set to 0 in the resize event.

cg2121 commented:
Updated to an actual fix of setting the correct spacing in the resize event.
